### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by AhmedTareek**
tests for item class

### CODE REVIEW

File: source/test/ItemTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @Test
+    void getName() {
+        assertEquals("pepsi",item0.getName());
+        assertEquals("fees",item1.getName());
+    }
+
+    @Test
+    void getPriceCaseNormal() {
```
Comment (ZiadAmerr) [inline source/test/ItemTest.java:27]:
```suggestion
    void testGetPrice() {
```

File: source/test/ItemTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        assertEquals(5,item0.getPrice());
+        assertEquals(99999,item1.getPrice());
+    }
+
+    @Test
+    void itemWithNegativePrice(){
```
Comment (ZiadAmerr) [inline source/test/ItemTest.java:33]: Can we merge this into testGetPrice?

File: source/test/ItemTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                new Item(Item.ItemType.ITEM,-23,"water"));
+    }
+
+    @Test
+    void getDatePurchased() {
+        User user = new User("Bruce","batman","ilovecatwoman");
+        Account account = new Account(user);
+        account.deposit(5000);
+        item0.purchase(account);
+        assertEquals(LocalDateTime.now(),item0.getDatePurchased());
```
Comment (ZiadAmerr) [inline source/test/ItemTest.java:44]: This should be a close assertion and not an equality assertion as time may pass after item was purchased and that the date-time now will be ahead by some milliseconds

**Review by ZiadAmerr — CHANGES_REQUESTED** [review]
Amazing work, please review the requested changes and we can merge right after

File: source/test/ItemTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        Account account = new Account(user);
+        account.deposit(5000);
+        assertTrue(item0.purchase(account));
+        assertFalse(item1.purchase(account));
+
+    }
+
+    @Test
+    void getType() {
+        assertEquals("ITEM",item0.getType());
```
Comment (ZiadAmerr) [inline source/test/ItemTest.java:68]:
```suggestion
        assertEquals(Item.ItemType.ITEM, item0.getType());
```

File: source/test/ItemTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        assertEquals("BILL",item1.getType());
```
Comment (ZiadAmerr) [inline source/test/ItemTest.java:69]:
```suggestion
        assertEquals(Item.ItemType.BILL, item1.getType());
```

Comment (ZiadAmerr) [inline source/test/ItemTest.java:69]:
```suggestion
        assertEquals(Item.ItemType.BILL, item1.getType());
```

**Review by ZiadAmerr — CHANGES_REQUESTED** [review]
Some more changes

File: source/test/ItemTest.java (hunk 5 above)
Comment (AhmedTareek) [reply source/test/ItemTest.java]: getType() return a string

Comment (ZiadAmerr) [reply source/test/ItemTest.java]: Yes, will modify it to return an ItemType enum

**Comment by ZiadAmerr** [comment]
lgtm

**Review by ZiadAmerr — APPROVED** [review]

**Pull request merged**